src/engine/strat_engine/backstore/blockdev/v2.rs

```diff
@@ -51,11 +51,17 @@ pub fn integrity_meta_space(
     total_space: Sectors,
     journal_size: Sectors,
     block_size: Bytes,
-    tag_size: Bytes,
+    tag_size: IntegrityTagSpec,
 ) -> Sectors {
     Bytes(4096).sectors()
         + journal_size
-        + Bytes::from(((*total_space.bytes() / *block_size) * *tag_size + 4095) & !4095).sectors()
+        + Bytes::from(
+            (((((total_space.bytes() / block_size) * u128::from(tag_size.as_bits())) / 8 + 7)
```

Comment:
I'm struggling to understand the logic of rounding up to the nearest byte if we're representing this as bits. This seems to defeat the purpose of representing it as bits in the first place. Could you explain a little bit more what you're trying to do here?

Reply:
The 4095 in the expression '+4095 & !4095' is implicitly in units of bytes. If I don't round up, then the value for the tag space may underestimate by 7 the number of bits required. If, by a weird chance, that underestimate is also divisible by 4096, then the total allocation for integrity metadata is nicely aligned but up to 7 bits short of what might be required.

Reply:
I think this is another reason why the units would make more sense in bytes. The best case scenario is that we end up overestimating in a way that rounds to Po8 hash sizes no matter what so this does not seem like we are getting anything from representing it as bits.

Reply:
I would be happy to change the order of operations. Instead  of multiplying the number of blocks  by the bits value and then rounding up to the nearest byte, rounding the bits value up to the nearest byte and multiplying it by the number of blocks. Of course, that has no effect at this point on the computation's result.

What I think would be shortsighted at this point is  limiting our D-Bus API and our pool-level  metadata to allow only units of bytes.

I'll make the proposed change.